### ISSUE TITLE
SNOW-1009569: snow connection add fails - cannot write config.toml

### ISSUE THREAD

**Issue by ahijstek** [Snowflake-Labs/snowcli#658]
### SnowCLI version

2.0.0a3

### Python version

3.10

### Platform

MacOS Sonoma 14.2.1 - Intel

### What happened

Hi all - today I started the 

Intro to Snowpark Container Services today
https://quickstarts.snowflake.com/guide/intro_to_snowpark_container_services/index.html?index=..%2F..index#2 

In there you need to download snowcli and add a connection via

 snow connection add
After filling out the relevant variables I get:

[Errno 2] No such file or directory: ‘/Users/xxxxxx/Library/Application Support/snowflake/config.toml’
Note that
 /Users/xxxxxx/Library/Application Support
 does exist - but the directory 'snowflake' thereunder does -not- exist. 

I added this snowflake directory myself and I retried
 snow concection add 
again - this does work! Not sure why snow assumes this directory to be there?

Thanks.

### Console output

```Shell
[Errno 2] No such file or directory: ‘/Users/xxxxxx/Library/Application Support/snowflake/config.toml’
I replaced my username with xxxxxx

After adding snowflake directory myself I get:
Wrote new connection CONTAINER_hol to /Users/xxxxxx/Library/Application Support/snowflake/config.toml
which is perfect.

NB: I replaced my username with xxxxxx
```

**Comment by sfc-gh-astus** [comment]
It will be fixed in #666

**Comment by sfc-gh-astus** [comment]
Fix merged
